### ISSUE TITLE
Destructuring in fn arguments warns for prop-types in 4.0.0-rc.0

### ISSUE THREAD

**Issue by lukekarrys** [yannickcr/eslint-plugin-react#443]
Here's a minimal example:

```js
const React = require('react');

module.exports = React.createClass({
  method1() {
    // Destructuring warns for
    // 'length' is missing in props validation  react/prop-types
    ['one', 'two', 'three'].forEach(({length}) => {
      console.log(length)
    })
  },

  method2() {
    // This doesnt warn
    ['one', 'two', 'three'].forEach((item) => {
      console.log(item.length)
    })
  }
})
```

Tested with and without `babel-eslint`. Here's my eslintrc without babel-eslint:

```
{
  "parserOptions": {
    "ecmaVersion": 6
  },

  "env": {
    "es6": true
  },

  "plugins": [
    "react"
  ],

  "extends": "plugin:react/recommended"
}
```

Also heres a gist with all the code: https://gist.github.com/lukekarrys/d3fa81ff63ea0cf1cf3a

**Comment by lukekarrys** [comment]
Looked at this for a little bit, but the best I could do was a failing test. Wasn't sure if this was worth opening a PR since its pretty simple but the failing test is in my fork: https://github.com/lukekarrys/eslint-plugin-react/commit/ca21391849ef99bb70db54e28df6c0fb783ce269

**Comment by yannickcr** [comment]
Thanks, I'll have a look.
